featurebyte/session/base.py

```diff
@@ -833,6 +855,19 @@ def remove_materialized_tables(cls, table_names: list[str]) -> list[str]:
                 out.append(table_name)
         return out
 
+    async def list_droppable_views_in_working_schema(self) -> list[str]:
+        """
+        List views in the working schema that can be dropped without losing data. These are the
+        views that will be reinstated by WorkingSchemaService when recreating the working schema.
+
+        Returns
+        -------
+        list[str]
+        """
+        views = await self.session.list_views(self.session.database_name, self.session.schema_name)
+        views_names = [view.name for view in views]
+        return self.remove_materialized_tables(views_names)
```

Comment:
Why do we need this? So far all the views we create are for historical feature tables, and we don't want to be dropping such views.

Reply:
Following is happening: 
- Some integration tests are triggering https://github.com/featurebyte/featurebyte/blob/2c0433366d9308a9fbf92dea11d47906e6dac92d/featurebyte/service/working_schema.py#L32 
- This lead (based on the DWH) lead to dropping tables, e.g. for Snowflake this code https://github.com/featurebyte/featurebyte/blob/2c0433366d9308a9fbf92dea11d47906e6dac92d/featurebyte/session/snowflake.py#L405 
- Both snowflake and spark return views as a part of `list_tables` response, but neither can drop view via "DROP TABLE" statement. 
- This lead to failures like "DROP TABLE expects object Table but View received" or something like that.  

That's why I add new function, which drops views separately. 

Regarding whether we want to drop historical feature tables or not - this is a good question. Historical tables are definitively getting dropped right now (being tables). Implementation in master just lists all tabes in schema as far as I can see, it is not very picky on the semantic meaning of the table.

Reply:
We seem to drop everything except tables which start with `MaterializedTableNamePrefix` prefix. 
Approach to creating a historical feature table name hasn't changed 
https://github.com/featurebyte/featurebyte/blob/2c0433366d9308a9fbf92dea11d47906e6dac92d/featurebyte/worker/task/historical_feature_table.py#L58

but in some reasons it wanted to drop views

Reply:
I guess I need to investigate this a bit more. Looking at the code - those views aren't supposed to be dropped.

Reply:
Solved, turned out this test have seen some other views created by feature table cache test and was trying to drop those.